raft/src/main/java/org/apache/kafka/snapshot/SnapshotWriter.java

```diff
@@ -85,6 +85,20 @@ public OffsetAndEpoch snapshotId() {
         return snapshot.snapshotId();
     }
 
+    /**
+     * Returns the last log offset which is represented in the snapshot.
+     */
+    public long lastOffsetFromLog() {
```

Comment:
One of the users of this API had some confusion with this offset and the offsets return in the `Batch` by the `SnapshotReader` iterator. I wanted to make it clear that this offset and epoch refers to the offset and epoch found in the `ReplicatedLog` or `handleCommit`.

While the offsets reported by the `Batch` for the `SnapshotReader` iterator are unrelated to the log's offsets.

How about `lastContainedLogOffset` and `lastContainedLogEpoch`?

Reply:
Yeah, makes sense. I was sort of considering if it would be useful to have a `SnapshotId` object. Currently we use `OffsetAndEpoch` in other cases, but maybe a separate object would let us have better names. It would also let us define inclusive and exclusive methods.

Reply:
I am up for that. Do you mind if I file a jira for this?